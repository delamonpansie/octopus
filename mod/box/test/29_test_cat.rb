#!/usr/bin/ruby

$: << File.dirname($0) + '/lib'
require 'run_env'

class Env < RunEnv
  def config
    super + <<EOD
object_space[0].enabled = 1
object_space[0].index[0].type = "TREE"
object_space[0].index[0].unique = 1
object_space[0].index[0].key_field[0].fieldno = 0
object_space[0].index[0].key_field[0].type = "STR"
object_space[0].index[0].key_field[1].fieldno = 1
object_space[0].index[0].key_field[1].type = "STR"
EOD
  end
end

Env.connect_eval do |env|
  insert %w[1 2 3]
  delete %w[1 2]
  select %w[1]

  insert %w[1 2 3]
  insert %w[1 2a 3]
  log_try { update_fields %w[1], [1, :set, "aa"] }
  update_fields %w[1 2], [1, :set, "aa"]
  select %w[1]

  env.stop
  puts `./octopus --cat 00000000000000000002.xlog`.gsub(/ tm:\d+\.\d+ /, ' ')
end
