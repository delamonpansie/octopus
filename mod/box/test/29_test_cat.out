# box.insert(["1", "2", "3"])
1

# box.delete(["1", "2"])
1

# box.select(["1"])
[]

# box.insert(["1", "2", "3"])
1

# box.insert(["1", "2a", "3"])
1

# box.update_fields(["1"], [1, :set, "aa"])
Failed with: {code: 0x202, message: 'key fields count doesn't match'}
# box.update_fields(["1", "2"], [1, :set, "aa"])
2

# box.select(["1"])
[["1", "2a", "3"], ["1", "aa", "3"]]

lsn:2 shard:0 scn:2 t:wal/usr13 INSERT n:0 flags:00000000 <"1", "2", "3">
lsn:3 shard:0 scn:3 t:wal/usr20 DELETE_1_3 n:0 <"1", "2">
lsn:4 shard:0 scn:4 t:wal/usr13 INSERT n:0 flags:00000000 <"1", "2", "3">
lsn:5 shard:0 scn:5 t:wal/usr13 INSERT n:0 flags:00000000 <"1", 24882:"2a", "3">
lsn:6 shard:0 scn:6 t:wal/usr19 UPDATE_FIELDS n:0 flags:00000000 <"1", "2"> [field_no:1 op:set <24929:"aa">] 
